### PR TITLE
Makes the master controller's auto-restart work

### PR DESCRIPTION
**Pull request by Crazylemon64**
When something goes horribly in a controller, it gets deleted, and the global variable it corresponds to is left empty. This is an attempt to remedy this so if something important dies, it can be restarted as appropriate.

However for some weird reason the mob controller isn't working on this branch 🤔

### CODE REVIEW

File: code/__DEFINES/process_scheduler.dm [hunk 1]
```diff
@@ -18,4 +18,17 @@
 #define SCHECK if(world.tick_usage >= next_sleep_usage) defer()
 
 //Timing Controller
-#define GLOBAL_PROC	"some_magic_bullshit"
\ No newline at end of file
+#define GLOBAL_PROC	"some_magic_bullshit"
```
Comment (SamHPurp) [inline code/__DEFINES/process_scheduler.dm]: 😆

File: code/controllers/Processes/air.dm [hunk 2]
```diff
@@ -50,6 +50,11 @@ var/global/datum/controller/process/air_system/air_master
 	stat(null, "[last_active] active")
 	stat(null, "[last_excited] EG | [last_hpd] HPD | [last_asc] ASC | [last_hotspots] Hot")
 
+/datum/controller/process/air_system/copyStateFrom()
+	air_master = src
```
Comment (Crazylemon64) [inline code/controllers/Processes/air.dm]: oh

File: code/__DEFINES/process_scheduler.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/__DEFINES/process_scheduler.dm]: leave it to @RemieRichards

**Comment by Crazylemon64** [comment]
Hey does anyone have advice on this or is this going to go the way of my firebug PR

File: code/controllers/Processes/mob.dm [hunk 3]
```diff
@@ -33,4 +33,7 @@ var/global/datum/controller/process/mob/mob_master
 		else
 			catchBadType(M)
 			mob_list -= M
-	current_cycle++
\ No newline at end of file
+	current_cycle++
+
+
+DECLARE_GLOBAL_CONTROLLER(mob, mob_master)
```
Comment (Fox-McCloud) [inline code/controllers/Processes/mob.dm]: check line #13 as to maybe why it's not working.

**Comment by Allfd** [comment]
👎 x 💯 

This is a **_major_** buff to the process controller.

**Comment by taukausanake** [comment]
@Crazylemon64 You had a firebug PR? Never saw it and I have alerts for the whole repo turned on

**Comment by Crazylemon64** [comment]
@taukausanake https://github.com/ParadiseSS13/Paradise/pull/5630

File: code/controllers/Processes/mob.dm (hunk 3 above)
Comment (Crazylemon64) [inline code/controllers/Processes/mob.dm]: Ah, thanks!

**Comment by Crazylemon64** [comment]
This should work now!

**Comment by Crazylemon64** [comment]
Should work now @Fox-McCloud

**Comment by fludd12** [comment]
I knew I wasn't insane that one time I mentioned the MC exploding.
I was just wrong about everything else :+1:

File: code/controllers/Processes/air.dm [hunk 4]
```diff
@@ -147,3 +149,5 @@ var/global/datum/controller/process/air_system/air_master
 	icemaster.icon_state = "snowfloor"
 	icemaster.layer = TURF_LAYER+0.1
 	icemaster.mouse_opacity = 0
+
+DECLARE_GLOBAL_CONTROLLER(air_system, air_master)
```
Comment (Fox-McCloud) [inline code/controllers/Processes/air.dm]: You have this in here twice, why is that?

File: code/controllers/Processes/mob.dm [hunk 5]
```diff
@@ -8,8 +8,6 @@ var/global/datum/controller/process/mob/mob_master
 	schedule_interval = 20 // every 2 seconds
 	start_delay = 16
 	log_startup_progress("Mob ticker starting up.")
-	if(mob_master)
-		qdel(mob_master) //only one mob master
 	mob_master = src
```
Comment (Fox-McCloud) [inline code/controllers/Processes/mob.dm]: Shouldn't this line be removed?

File: code/controllers/Processes/shuttles.dm [hunk 6]
```diff
@@ -72,6 +72,9 @@ var/const/CALL_SHUTTLE_REASON_LENGTH = 12
 			continue
 		mobile.Remove(thing)
 
+
```
Comment (Fox-McCloud) [inline code/controllers/Processes/shuttles.dm]: shouldn't line 43 be removed?

File: code/controllers/Processes/air.dm (hunk 4 above)
Comment (Crazylemon64) [inline code/controllers/Processes/air.dm]: I may have overlooked this, it wasn't there the last time I looked at it

File: code/controllers/Processes/mob.dm (hunk 5 above)
Comment (Crazylemon64) [inline code/controllers/Processes/mob.dm]: gah

**Comment by Kyep** [comment]
@Fox-McCloud @Crazylemon64 @Necaladun @Allfd @FreeStylaLT Can we please get this merged ASAP? After hopping on today, I saw two rounds in a row, both of which had to be force-restarted early, because of freezes to mob processing that this PR would fix.

Having to force-restart occasionally is annoying, but having to force-restart multiple rounds in a row is really annoying to players.

**Comment by Necaladun** [comment]
I barely understand any of this, so not my call

**Pull request merged**